Modify S3257: Migrate to LayC (#3267)
SonarSource/rspec: rules/S3257/javascript/rule.adoc

== Why is this an issue?

> TypeScript has a feature called "type inference" which allows it to automatically infer the type of a variable based on its initial value. This means that if you initialize a variable with a certain type of value, TypeScript will assume that this variable should always hold that type of value.

Reviewer: ```suggestion
TypeScript supports type inference, a mechanism that automatically infers the type of a variable based on its initial value. This means that if you initialize a variable with a particular value, TypeScript will assume that this variable should always hold that type of value.
```

> Unnecessarily verbose declarations and initializations make it harder to read the code and should be simplified. Specifically, primitive (``++number++``, ``++string++``, ``++boolean++`` and others) types should be omitted from variable and parameter declaration when they can be easily inferred from the initialized or defaulted value.

Reviewer: ```suggestion
Unnecessarily verbose declarations and initializations make it harder to read the code and should be simplified. Specifically, primitive (``++number++``, ``++string++``, ``++boolean++`` and others) types should be omitted from variable and parameter declarations when they can be easily inferred from the initialized or defaulted value.
```

> Unnecessarily verbose declarations and initializations make it harder to read the code and should be simplified. Specifically, primitive (``++number++``, ``++string++``, ``++boolean++`` and others) types should be omitted from variable and parameter declaration when they can be easily inferred from the initialized or defaulted value.

Reviewer: If the rule isn't limited to primitive types, I would reword this as follows:

```suggestion
Unnecessarily verbose declarations and initializations make it harder to read the code and should be simplified. Therefore, type annotations should be omitted from variable and parameter declarations when they can be easily inferred from the initialized or defaulted value.
```

== How to fix it

> When the type can be easily inferred from the context, omit the type declaration to make the code more concise and readable.

Reviewer: ```suggestion
Omit explicit type annotations in declarations whenever the type can be inferred from the context.
```
